p4-16/psa/psa.p4

```diff
@@ -125,8 +125,8 @@ struct psa_egress_output_metadata_t {
   // Egress control block begins executing.
   bool                     clone;         // false
   ClassOfService_t         clone_class_of_service; // 0
+  bool                     recirculate;   // set by environment
   bool                     drop;          // false
-  bool                     recirculate;   // false
```

Comment:
We can discuss in the PSA WG meeting, but I think it would be preferable to either:

+ remove the 'recirculate' flag entirely, relying on somehow getting the value of egress_port as an 'in' parameter to the egress deparser, and defining psa_recirculate() to return the value (egress_port == PORT_RECIRCULATE)

or

+ add a new 'in' parameter to the deparser that is a struct containing the recirculate flag.  If it should not be modifiable by the egress control block, it shouldn't be part of an out or inout parameter of the egress control block.

Reply:
I prefer to remove `recirculate` flag and replace with an architecture defined `psa_recirculate()`. But, passing the entire `psa_egress_input_metadata_t` as `in` parameter to egress deparser is a bit too much. We may need yet another parameter to egress deparser.